packages/common/src/ContractInteractor.ts

```diff
@@ -361,6 +385,46 @@ export class ContractInteractor {
     return await this.IRelayRegistrar.at(address)
   }
 
+  async _createERC20 (address: Address): Promise<IERC20TokenInstance> {
+    return await this.IERC20Token.at(address)
+  }
+
+  /**
+   * Queries the balance of the token and displays it in a human-readable format, taking 'decimals' into account.
+   * Note: does not round up the fraction and truncates it.
+   */
+  async getTokenBalanceFormatted (address: Address): Promise<string> {
+    const balance = await this.erc20Token.balanceOf(address)
+    return await this.formatTokenAmount(balance)
+  }
+
+  async formatTokenAmount (balance: BN): Promise<string> {
+    const { tokenSymbol, tokenDecimals } = await this.getErc20TokenMetadata()
+    return formatTokenAmount(balance, tokenDecimals, tokenSymbol)
+  }
+
+  async getErc20TokenMetadata (): Promise<ERC20TokenMetadata> {
+    let tokenName: string
+    try {
+      tokenName = await this.erc20Token.name()
```

Comment:
does this work on real network? I remember I failed to call view functions unless given a real `from` address (with balance), or `gasPrice:0`